packages/lib-utils/src/k8s/k8s-utils.ts

```diff
@@ -50,3 +62,189 @@ export const getK8sResourceURL = (
 
   return `/${resourcePath}`;
 };
+
+const requirementToString = (requirement: MatchExpression): string => {
+  if (requirement.operator === 'Equals') {
+    return `${requirement.key}=${requirement.values?.[0]}`;
+  }
+
+  if (requirement.operator === 'NotEquals') {
+    return `${requirement.key}!=${requirement.values?.[0]}`;
+  }
+
+  if (requirement.operator === 'Exists') {
+    return requirement.key;
+  }
+
+  if (requirement.operator === 'DoesNotExist') {
+    return `!${requirement.key}`;
+  }
+
+  if (requirement.operator === 'In') {
+    return `${requirement.key} in (${_.toArray(requirement.values).join(',')})`;
+  }
+
+  if (requirement.operator === 'NotIn') {
+    return `${requirement.key} notin (${_.toArray(requirement.values).join(',')})`;
+  }
+
+  if (requirement.operator === 'GreaterThan') {
+    return `${requirement.key} > ${requirement.values?.[0]}`;
+  }
+
+  if (requirement.operator === 'LessThan') {
+    return `${requirement.key} < ${requirement.values?.[0]}`;
+  }
+
+  return '';
+};
+
+const createEquals = (key: string, value: string): MatchExpression => ({
+  key,
+  operator: 'Equals',
+  values: [value],
+});
+
+const isOldFormat = (selector: Selector | MatchLabels) =>
```

Comment:
@andrewballantyne @vojtechszocs I can remove this logic, but until we test how this behaves thoroughly I figured I'd leave it.